src/storage/storage_handler.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+
+from src.utils.config import ConfigArguments
+
+class DataStorage(ABC):
+    def __init__(self):
+        self._args = ConfigArguments.get_instance()
+    
+    @abstractmethod
+    def get_uri(self, file_name):
+        """
+            This method returns URI of a file name based on the implemented file system.
+            eg: For a file in S3, s3:// has to be prefixed to the file name.
+            eg: For a file in hdfs, hdfs:// has to be prefixed to the file name.
+        """
+        pass
+
+    @abstractmethod
+    def makedirs(self, dir_name, exist_ok=False):
+        """
+            This method creates a directory in the backend system.
+            If exist_ok flag is set, no error is returned if dir is already present.
+        """
+        pass
+
+    @abstractmethod
+    def listdir(self, dir_name):
+        """
+            This method lists all files in the directory.
+        """
+        pass
+
+    @abstractmethod
+    def exists(self, path):
+        """
+            This method checks if path already exists.
+        """
+        pass
+
+    @abstractmethod
+    def rmtree(self, dir_name):
+        """
+            This method removes the directory recursively.
+        """
+        pass
+
+    @abstractmethod
+    def isfile(self, file_name):
```

Comment:
Instead of checking for is file, we should get type of node and the client of this call can check the type of node as file/object/namespace